clang/test/CodeGenCXX/ptrauth-throw.cpp

```diff
@@ -0,0 +1,20 @@
+// RUN: %clang_cc1 -triple arm64-apple-ios -fptrauth-calls -fcxx-exceptions -emit-llvm %s -o - | FileCheck %s
+
+class Foo {
+ public:
+  ~Foo() {
+  }
+};
+
+// CHECK-LABEL: define void @_Z1fv()
+// CHECK:  call void @__cxa_throw(ptr %{{.*}}, ptr @_ZTI3Foo, ptr ptrauth (ptr @_ZN3FooD1Ev, i32 0))
+void f() {
+  throw Foo();
+}
+
+// __cxa_throw is defined to take its destructor as "void (*)(void *)" in the ABI.
+// CHECK-LABEL: define void @__cxa_throw({{.*}})
+// CHECK:  call void {{%.*}}(ptr noundef {{%.*}}) [ "ptrauth"(i32 0, i64 0) ]
```

Comment:
When #96992 is merged, it's probably worth adding a RUN line with `-fptrauth-function-pointer-type-discrimination` and ensure that type discrimination is applied properly. It's not a blocker for merging this PR, but it would be nice to have such a test in future.

Reply:
Agreed, I had to actually remove that from the test because that PR had not landed